File: src/main/java/ee/carlrobert/codegpt/settings/service/ollama/OllamaSettingsState.java
package ee.carlrobert.codegpt.settings.service.ollama;

import java.util.Objects;

public class OllamaSettingsState {

    private String host = "http://localhost:11437";
    private String model = "";
    private boolean codeCompletionsEnabled = true;
    private int codeCompletionMaxTokens = 128;

    public String getHost() {
        return host;
    }

    public void setHost(String host) {
        this.host = host;
    }

    public String getModel() {
        return model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public boolean isCodeCompletionsEnabled() {
        return codeCompletionsEnabled;
    }

    public void setCodeCompletionsEnabled(boolean codeCompletionsEnabled) {
        this.codeCompletionsEnabled = codeCompletionsEnabled;
    }

    public int getCodeCompletionMaxTokens() {
        return codeCompletionMaxTokens;
    }

    public void setCodeCompletionMaxTokens(int codeCompletionMaxTokens) {
        this.codeCompletionMaxTokens = codeCompletionMaxTokens;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        OllamaSettingsState that = (OllamaSettingsState) o;
        return Objects.equals(host, that.host)
                && Objects.equals(model, that.model)
                && codeCompletionsEnabled == that.codeCompletionsEnabled
                && codeCompletionMaxTokens == that.codeCompletionMaxTokens;
    }

    @Override
    public int hashCode() {
        return Objects.hash(host, model, codeCompletionsEnabled, codeCompletionMaxTokens);
    }
}

File: src/main/kotlin/ee/carlrobert/codegpt/settings/service/ollama/OllamaSettingsState.kt
package ee.carlrobert.codegpt.settings.service.ollama

import ee.carlrobert.codegpt.codecompletions.ollama.OllamaInlineCompletionModel

data class OllamaSettingsState(
    val host: String = "http://localhost:11434",
    val model: String = "",
    val codeCompletionMaxTokens: Int = 128,
    val codeCompletionsEnabled: Boolean = true,
) {
    val fimTemplate = OllamaInlineCompletionModel.entries.firstOrNull { model.contains(it.identifier) }?.fimTemplate

    fun isCodeCompletionPossible(): Boolean {
        return fimTemplate != null && codeCompletionsEnabled
    }
}
